presidio-analyzer/presidio_analyzer/analyzer_engine.py

```diff
@@ -55,30 +60,69 @@ def __init__(
             app_tracer = AppTracer()
         self.app_tracer = app_tracer
 
-        self.default_language = default_language
+        self.supported_languages = supported_languages
 
         self.nlp_engine = nlp_engine
         self.registry = registry
 
         # load all recognizers
         if not registry.recognizers:
-            registry.load_predefined_recognizers()
+            registry.load_predefined_recognizers(
+                nlp_engine=self.nlp_engine, languages=self.supported_languages
+            )
 
         self.enable_trace_pii = enable_trace_pii
         self.default_score_threshold = default_score_threshold
 
+    @staticmethod
+    def _create_nlp_engine(nlp_conf_file_path: Path) -> NlpEngine:
```

Comment:
would extract it to a different class (NlpEngineProvider) or file (since static). Would make the classes and tests less complex, engine not handle files. Can be along with create_nlp_engine as well

Reply:
Did exactly this. Thanks for the suggestion.